confidant/authnz/userauth.py

```diff
@@ -270,6 +272,85 @@ def log_in(self):
         raise NotImplementedError
 
 
+class HeaderAuthenticator(AbstractUserAuthenticator):
+    """
+    User authenticator that pulls user information from HTTP headers.
+    Note that this assumes we're running behind some form of load-balancer or
+    reverse proxy that performs the authentication, and that simply being able
+    to make requests to this service implies that the user is authenticated.
+    """
+
+    def __init__(self):
+        self.username_header = app.config['HEADER_AUTH_USERNAME_HEADER']
+        self.email_header = app.config['HEADER_AUTH_EMAIL_HEADER']
+        self.first_name_header = app.config.get('HEADER_AUTH_FIRST_NAME_HEADER')
+        self.last_name_header = app.config.get('HEADER_AUTH_LAST_NAME_HEADER')
+
+    @property
+    def auth_type(self):
+        return 'header'
+
+    def assert_headers(self):
+        """Asserts that the current request contains the appropriate headers."""
+        if self.username_header not in request.headers:
+            raise errors.UserUnknownError('No username header in request')
+
+        if self.email_header not in request.headers:
+            raise errors.UserUnknownError('No email header in request')
+
+    def current_user(self):
+        self.assert_headers()
+
+        info = {
+            'email': request.headers[self.email_header],
+
+            # TODO: should we use a string like "unknown", fall back to the
+            # email/username, ...?
```

Comment:
I think it's fine to leave these as empty strings, if they aren't supported as headers.